libraries/resource_providers/api200/server_profile_provider.rb

```diff
@@ -30,6 +30,29 @@ def set_connections(type, connection_list)
         true
       end
 
+      def build_volume_attachments
+        return unless @new_resource.volume_attachments
+        @new_resource.volume_attachments.each do |options|
+          options = convert_keys(options, :to_s)
+          volume_name = options['volume']
+          volume_data = options['volume_data']
+          attachment_data = options.fetch('attachment_data', {})
+          raise('To add volume attachments you need volume or volume_data into each data of volume_attachments') unless volume_name || volume_data
+          if volume_name
+            volume = load_resource(:Volume, name: volume_name)
+            @item.add_volume_attachment(volume, attachment_data)
+          else
+            storage_system_name = options.delete('storage_system')
+            storage_pool_name = options.delete('storage_pool')
+            raise('To create a new volume with attachment you need storage_system and storage_pool') unless storage_system_name && storage_pool_name
```

Comment:
(...) with **an** attachment you need **to specify** the 'storage_system' and 'storage_pool' names inside 'options'.